dbms/src/Flash/Planner/FinalizeHelper.cpp

```diff
@@ -0,0 +1,137 @@
+// Copyright 2022 PingCAP, Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <Common/FmtUtils.h>
+#include <Common/TiFlashException.h>
+#include <Core/Block.h>
+#include <Core/Names.h>
+#include <Core/NamesAndTypes.h>
+#include <Flash/Planner/FinalizeHelper.h>
+#include <Interpreters/ExpressionActions.h>
+#include <common/types.h>
+
+#include <unordered_map>
+
+namespace DB::FinalizeHelper
+{
+namespace
+{
+String namesToString(const Names & names)
```

Comment:
maybe  namesToString,schemaToString, blockMetaToString should be in Names.h, NamesAndTypes.h and Block.h